code/modules/robotics/robot/robot_docking_station.dm

```diff
@@ -246,6 +246,23 @@ TYPEINFO(/obj/machinery/recharge_station)
 	if (ishuman(AM))
 		src.move_human_inside(user, AM)
 
+/obj/machinery/recharge_station/receive_silicon_hotkey(mob/user)
+	. = ..()
+
+	if (!isAI(user))
+		return
+
+	var/mob/living/silicon/ai/mainframe = null
+	if (isAIeye(user))
+		var/mob/living/intangible/aieye/eye = user
+		mainframe = eye.mainframe
+	else
+		mainframe = user
+
+	if(user.client.check_key(KEY_OPEN))
+		if (src.occupant && !ishuman(src.occupant))
```

Comment:
Don't really need the `ishuman()` check - `deploy_to_shell()` has an `issilicon()` check at the top already